src/main/java/graphql/execution/FieldCollector.java

```diff
@@ -21,7 +35,45 @@ public FieldCollector() {
     }
 
 
-    public void collectFields(ExecutionContext executionContext, GraphQLObjectType type, SelectionSet selectionSet, List<String> visitedFragments, Map<String, List<Field>> fields) {
+    /**
+     * Given a list of fields this will collect the sub-field selections and return it as a map
+     *
+     * @param executionContext the {@link ExecutionContext} in play
+     * @param objectType       the graphql object type in context
+     * @param fields           the list of fields to collect for
+     *
+     * @return a map of the sub field selections
+     */
+    public Map<String, List<Field>> collectFields(ExecutionContext executionContext, GraphQLObjectType objectType, List<Field> fields) {
+        Map<String, List<Field>> subFields = new LinkedHashMap<>();
+        List<String> visitedFragments = new ArrayList<>();
+        for (Field field : fields) {
+            if (field.getSelectionSet() == null) {
+                continue;
+            }
+            this.collectFields(executionContext, objectType, field.getSelectionSet(), visitedFragments, subFields);
+        }
+        return subFields;
+    }
+
+    /**
+     * Given a selection set this will collect the sub-field selections and return it as a map
+     *
+     * @param executionContext the {@link ExecutionContext} in play
+     * @param objectType       the graphql object type in context
+     * @param selectionSet     the selection set to collect on
+     *
+     * @return a map of the sub field selections
+     */
+    public Map<String, List<Field>> collectFields(ExecutionContext executionContext, GraphQLObjectType objectType, SelectionSet selectionSet) {
+        Map<String, List<Field>> subFields = new LinkedHashMap<>();
+        List<String> visitedFragments = new ArrayList<>();
+        this.collectFields(executionContext, objectType, selectionSet, visitedFragments, subFields);
+        return subFields;
+    }
+
+
+    private void collectFields(ExecutionContext executionContext, GraphQLObjectType type, SelectionSet selectionSet, List<String> visitedFragments, Map<String, List<Field>> fields) {
```

Comment:
I can make this public again but I figured 3.x and breaking changes and its unlikely to break ANYONE really